server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1049,44 +1057,35 @@ private static boolean checkAllRemotePITShardsWereReturnedBySearchShards(
             .allMatch(searchContextIdForNode -> searchContextIdForNode.getClusterAlias() == null);
     }
 
-    Index[] resolveLocalIndices(OriginalIndices localIndices, ClusterState clusterState, SearchTimeProvider timeProvider) {
-        if (localIndices == null) {
-            return Index.EMPTY_ARRAY; // don't search on any local index (happens when only remote indices were specified)
-        }
-
-        List<String> frozenIndices = null;
-        Index[] indices = indexNameExpressionResolver.concreteIndices(clusterState, localIndices, timeProvider.absoluteStartMillis());
-        for (Index index : indices) {
-            IndexMetadata indexMetadata = clusterState.metadata().index(index);
-            if (indexMetadata.getSettings().getAsBoolean("index.frozen", false)) {
-                if (frozenIndices == null) {
-                    frozenIndices = new ArrayList<>();
-                }
-                frozenIndices.add(index.getName());
+    void frozenIndexCheck(ResolvedIndices resolvedIndices) {
```

Comment:
I kept the frozen index check in `TransportSearchAction` because it appears to be specific to search actions. LMK if that should change.